### ISSUE TITLE
Text Cell PDF Images Disappear

### ISSUE THREAD

**Issue by ijoseph** [HappenApps/Quiver#1251]
Thanks again for a great product. 

TL;DR: Towards the end of self-debugging, where, exactly, are PDF images stored in the file format (Quiver `json`s?) 

Long version: 

I have a bizarre issue: when I copy `PDF` images into Text Cells, everything's fine for awhile:

![image](https://user-images.githubusercontent.com/549191/50533340-827fb080-0adc-11e9-8b36-776ead1ee184.png)

Then, at some later point in time (anywhere between 15 minutes and a few weeks), the images seem to be missing (blue boxes with a question mark: 
![image](https://user-images.githubusercontent.com/549191/50533323-3b91bb00-0adc-11e9-956e-bf41922c0e71.png)
). 

This is maddening because it's hard to figure out exactly what caused the above. 

More info: this is particularly occurring when I get PDF vector graphics from LatexIt. e.g.: 
![image](https://user-images.githubusercontent.com/549191/50533335-63811e80-0adc-11e9-9d5f-e518126e94bf.png)

I have not observed this behavior with `PNG`s.

**Comment by ijoseph** [comment]
Okay, I did some digging. Apparently inserted `pdf` images result in something like: 

> `<img src=\"blob:file:///aa5d8482-fbc9-44cc-a307-c1c8112509ee\">`

Whereas inserted `png` images are actually, apparently, converted into `jpg`s:

> `<img src=\"quiver-image-url/B0D4EE5522CE7C8065CC6736BC4A412F.jpg\" width=\"13\" height=\"50\">`

Interestingly, I can find that file: `B0D4EE5522CE7C8065CC6736BC4A412F.jpg` in the `Quiver.qvlibrary` folder. However, I can *not* find anything matching the strange blob (`aa5d8482-fbc9-44cc-a307-c1c8112509ee`) above anywhere on my machine.

**Comment by ijoseph** [comment]
Update 2: okay, quitting and starting the app again seems sufficient to convert my PDF images into question-mark boxes. So the images must be stored in memory somewhere, somehow pointed to by `blob:file:///…`.

**Comment by ylian** [comment]
Thanks for the detailed information. PNG, JPEG and most other image types are automatically saved to file, but PDF image is an edge case that I probably didn't handle.

The behaviour you have observed so far is the standard behaviour when you copy an image of any type into a text cell. The system basically keeps the image in memory temporarily, so restarting the app would resulting in the question mark. PNGs and JPEGs work because I have handled those image types and save the image from memory to file and update the image src.

I will look into a fix.

**Comment by ijoseph** [comment]
Thanks for your lightning response. Makes sense. That’s interesting. So when pasted in, Quiver renders any image by reading it directly from memory? Interesting. So the `blob`notation is a memory address? Moot questions I guess because I cannot actually change the source code myself. I look forward to the fix. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ylian** [comment]
@ijoseph Actually most work is done by the system. When you paste in an image, the system renders the image in the text cell, then gives it a fake URL. Quiver detects that an image has loaded, checks that it has a fake url, saves the image to disk, then updates the fake URL to a Quiver image url. The problem lies in the image detection part, as I need to make sure it's an image that's pasted in. As I said, I handled most image types, but pdf images are a special case.
